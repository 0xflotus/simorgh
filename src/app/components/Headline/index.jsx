import styled from 'styled-components';
import { C_EBON, FF_NEWS_SANS_REG } from '../../lib/constants/styles';
import mediaQuery from '../../helpers/mediaQueries';

const Headline = styled.h1`
  color: ${C_EBON};
  font-family: ${FF_NEWS_SANS_REG};
  font-size: 1.75em;
  line-height: 2em;
  ${mediaQuery.smartPhoneOnly} {
    font-size: 2em;
    line-height: 2.25em;
  }
  ${mediaQuery.tabletOnly} {
    font-size: 3.25em;
    line-height: 3.5em;
  }
  ${mediaQuery.desktopOnly} {
    font-size: 2.75em;
    line-height: 3em;
  }
`;

export default Headline;
