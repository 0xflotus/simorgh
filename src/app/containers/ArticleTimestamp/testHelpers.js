import moment from 'moment-timezone';

const magnitudes = {
  days: 24 * 60 * 60 * 1000,
  hours: 60 * 60 * 1000,
  minutes: 60 * 1000,
  seconds: 1000,
  milliseconds: 1,
};

const applyTimeDifference = (timestamp, timeDifference, isAdding) => {
  let acc = timestamp;

  const keyNames = Object.keys(timeDifference);

  keyNames.forEach(diff => {
    if (isAdding) {
      acc += timeDifference[diff] * magnitudes[diff];
    } else {
      acc -= timeDifference[diff] * magnitudes[diff];
    }
  });

  return acc;
};

const subtractTimeDifference = (timestamp, timeDifference) =>
  applyTimeDifference(timestamp, timeDifference, false);

const addTimeDifference = (timestamp, timeDifference) =>
  applyTimeDifference(timestamp, timeDifference, true);

export const timestampGenerator = timeDifference =>
  subtractTimeDifference(Date.now(), timeDifference);

export const multipleSameDayTimestampsGenerator = ({ intervals, date }) => {
  /**
   * This function accepts an array of intervals that represent the time
   * difference between specific moments in the past and Date.now(), and
   * returns an array of timestamps for those moments.
   *
   * The earliest one is set to 12:00am local time so that as long as all
   * intervals are less than 24 hours the timestamps returned will all be on
   * the same day, according to the machine's timezone.
   *
   * Note, the first interval provided must be the largest. If a date is not
   * provided it will use the current date.
   */

  if (!intervals || !Array.isArray(intervals)) {
    throw new Error(
      'Invalid intervals argument passed to multipleSameDayTimestampsGenerator',
    );
  }

  const firstTimestamp = moment(date)
    .startOf('day')
    .valueOf();
  const lastTimestamp = addTimeDifference(firstTimestamp, intervals[0]);
  return intervals
    .map(interval => subtractTimeDifference(lastTimestamp, interval))
    .concat(lastTimestamp);
};

export const isBritishSummerTime = timestamp =>
  moment.tz(timestamp, 'Europe/London').isDST();
