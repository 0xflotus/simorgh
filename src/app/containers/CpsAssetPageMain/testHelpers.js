import { render, waitForDomChange } from '@testing-library/react';

const renderWithHelmet = component => {
  /*
   * Similar to this problem https://github.com/testing-library/react-testing-library/issues/402
   * This helper was created to solve the problem of rendering helmet/head contents in snapshots.
   * Pass in a component that uses helmet somewhere in the component tree.
   * The full html tree is returned which you can then use to snapshot helmet/head contents.
   */

  const { container } = render(component);
  const noop = () => {};
  const ARBITRARY_TIMEOUT = 100; // seems long enough for any dom mutations to occur
  const headElement = document.querySelector('head');

  headElement.innerHTML = ''; // clear out head mutations from previous tests

  return waitForDomChange({
    container: headElement,
    timeout: ARBITRARY_TIMEOUT,
  })
    .catch(noop) // handle a waitForDomChange timeout
    .then(mutationsList => {
      const headMutationDetected = mutationsList && mutationsList.length;

      if (headMutationDetected) {
        // helmet was probably used so we should get the full html

        const htmlElement = document.querySelector('html');

        const helmetElements = document.querySelectorAll(
          '[data-react-helmet="true"]',
        );

        const removeHelmetAttributes = el =>
          el.removeAttribute('data-react-helmet'); // remove react-helmet attribute noise from elements

        removeHelmetAttributes(htmlElement); // remove react-helmet attribute noise from elements

        Array.from(helmetElements).forEach(removeHelmetAttributes);

        return { container: document.querySelector('html') };
      }

      return { container };
    });
};

export default (component, done) => {
  const removeWrappingDiv = container => container.firstChild;
  renderWithHelmet(component).then(({ container }) => {
    const hasOneChild = container.children.length === 1;
    /*
     * if the container has more than one child then it's a component that uses a
     * fragment at the top level so we should not select the first child because it
     * wouldn't snapshot the whole component
     */
    expect(
      hasOneChild ? removeWrappingDiv(container) : container,
    ).toMatchSnapshot();

    done();
  });
};
