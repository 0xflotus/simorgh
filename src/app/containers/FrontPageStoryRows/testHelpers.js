import { pathOr } from 'ramda';
import uuid from 'uuid';
import fixture from '#data/pidgin/frontpage';
import rtlFixture from '#data/urdu/frontpage';

const getFixtureData = dir => (dir === 'ltr' ? fixture : rtlFixture);

const getPromoFixtures = dir =>
  pathOr(null, ['content', 'groups'], getFixtureData(dir))
    .flatMap(group => pathOr(null, ['items'], group))
    .filter(item => pathOr(null, ['assetTypeCode'], item) === 'PRO')
    .map(item => ({ id: uuid(), ...item }));

export default getPromoFixtures;
