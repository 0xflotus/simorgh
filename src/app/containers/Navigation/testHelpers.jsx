import React from 'react';

export const dropdownTestId = 'dropdown';
export const scrollableTestId = 'scrollable-list';

export const scrollableListItems = (
  <ul data-testid={scrollableTestId}>
    <li>List Items</li>
  </ul>
);

export const dropdownListItems = (
  <div data-testid={dropdownTestId}>Dropdown</div>
);
