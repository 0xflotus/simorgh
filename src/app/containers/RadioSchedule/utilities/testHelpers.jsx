import React from 'react';
import RadioScheduleContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

const localRadioScheduleEndpoint = service => {
  const localhostURL = 'http://localhost:7080';
  const localServiceURL = `${localhostURL}/${service}`;
  return `${localServiceURL}/bbc_${service}_radio/schedule.json`;
};

const getToggleState = enabled => ({
  local: { radioSchedule: { enabled } },
  test: { radioSchedule: { enabled } },
});

/* eslint-disable react/prop-types */
const RadioSchedulesWithContext = ({
  service,
  radioScheduleToggle = false,
}) => (
  <ToggleContext.Provider
    value={{ toggleState: getToggleState(radioScheduleToggle) }}
  >
    <RequestContextProvider
      isAmp={false}
      pageType="frontPage"
      service={service}
      pathname={`/${service}`}
    >
      <ServiceContextProvider service={service}>
        <RadioScheduleContainer
          endpoint={localRadioScheduleEndpoint(service)}
        />
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */

export default RadioSchedulesWithContext;
