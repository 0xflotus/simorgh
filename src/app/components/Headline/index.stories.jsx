import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import Headline from './index';

storiesOf('Headline', module).add('default', () => (
  <Headline>This is my headline.</Headline>
));
