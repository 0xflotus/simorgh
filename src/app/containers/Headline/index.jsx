import React from 'react';
import { extractText } from '../../helpers/blockHandlers';
import { textDefaultPropTypes } from '../../models/proptypes';
import { headlineModelPropTypes } from '../../models/propTypes/headline';
import Headline from '../../components/Headline';

const HeadlineContainer = ({ blocks }) => {
  const { text } = extractText(blocks);

  if (!text) {
    return null;
  }

  return <Headline>{text}</Headline>;
};

HeadlineContainer.propTypes = headlineModelPropTypes;

HeadlineContainer.defaultProps = textDefaultPropTypes;

export default HeadlineContainer;
