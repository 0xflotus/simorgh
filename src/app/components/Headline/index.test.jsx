import React from 'react';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
import Headline from './index';

describe('Headline', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <Headline>This is text with no formatting.</Headline>,
  );
});
