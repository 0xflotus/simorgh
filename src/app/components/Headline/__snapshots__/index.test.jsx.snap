// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline should render correctly 1`] = `
.c0 {
  color: #222222;
  font-family: ReithSansNewsRegular,Arial,Helvetica,freesans,sans-serif;
  font-size: 2em;
}

<h1
  className="c0"
>
  This is text with no formatting.
</h1>
`;
